### PR TITLE
Add method to get undocumented job graph

### PR DESCRIPTION
**Pull request by keshavdv**
Also fixes tests to work with boot2docker/docker-machine.

### CODE REVIEW

File: itests/steps/chronos_steps.py [hunk 1]
```diff
@@ -53,6 +53,11 @@ def not_see_job_with_spaces(context, job_name):
     job_names = [job['name'] for job in jobs]
     assert 'test chronos job with spaces' not in job_names
 
+@then(u'we should be able to see {num_jobs} in the job graph')
+def not_see_job_with_spaces(context, num_jobs):
```
Comment (Rob-Johnson) [inline itests/steps/chronos_steps.py]: it doesn't look like you're using `num_jobs` here?

File: chronos/__init__.py [hunk 2]
```diff
@@ -153,7 +157,8 @@ def _check(self, resp, content):
             try:
                 payload = json.loads(content)
             except ValueError:
-                self.logger.error("Response not valid json: %s" % content)
+                if expect_json:
+                    self.logger.error("Response not valid json: %s" % content)
```
Comment (Rob-Johnson) [inline chronos/__init__.py:160]: how about using the Content-Type header here rather than requiring a parameter? Chronos seems to set it correctly

Comment (keshavdv) [inline chronos/__init__.py:160]: Ah, much better!

**Comment by Rob-Johnson** [comment]
shipit!

**Pull request merged**